lib/TPP/ConvertPackOptimization.cpp

```diff
@@ -0,0 +1,223 @@
+//===ConvertPackOptimization.cpp --------------------------------*----- C++-*-===//
+////
+//// Part of the LLVM Project, under the Apache License v2.0 with LLVM
+/// Exceptions. / See https://llvm.org/LICENSE.txt for license information. /
+/// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+////
+////===----------------------------------------------------------------------===//
+//
+//
+#include "TPP/BuilderUtils.h"
+#include "TPP/Dialect/Tpp/TppOps.h"
+#include "TPP/Dialect/Tpp/TppUtils.h"
+#include "TPP/Passes.h"
+#include "TPP/TransformUtils.h"
+#include "TPP/Transforms.h"
+#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Linalg/Utils/Utils.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Dialect/Traits.h"
+#include "mlir/Dialect/Utils/IndexingUtils.h"
+#include "mlir/Dialect/Utils/StaticValueUtils.h"
+#include "mlir/Support/MathExtras.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/ADT/TypeSwitch.h"
+using namespace mlir;
+using namespace mlir::tpp;
+
+#define GEN_PASS_CLASSES
+#include "TPP/Passes.h.inc"
+
+namespace {
+
+typedef struct MatchResult {
+  bool haveMatch;
+  int index;
+} MatchResult;
+
+SmallVector<MatchResult> innerDimMatchesIndex(tensor::PackOp packOp) {
```

Comment:
I am a bit lost here. What are you trying to do?

Reply:
Added a line of comment in code. Lemme know if that helps